packages/react-charting/src/components/SankeyChart/SankeyChart.base.tsx

```diff
@@ -105,38 +328,393 @@ export class SankeyChartBase extends React.Component<
+   * provide color to  individual node.
+   */
+  private _assignNodeColors() {
+    let colors: string[];
+    let borders: string[];
+    if (this.props.colorsForNodes && this.props.borderColorsForNodes) {
+      colors = this.props.colorsForNodes;
+      borders = this.props.borderColorsForNodes;
+    } else {
+      colors = DEFAULT_NODE_COLORS;
+      borders = DEFAULT_NODE_BORDER_COLORS;
+    }
+    let currentIndex = 0;
+    this.props.data.SankeyChartData!.nodes.forEach((node: any) => {
+      if (!node.color && !node.borderColor) {
+        node.color = colors[currentIndex];
+        node.borderColor = borders[currentIndex];
+      } else if (node.color && !node.borderColor) {
+        node.borderColor = '#757575';
+      } else if (node.borderColor && !node.color) {
+        node.color = '#F5F5F5';
+      }
+      currentIndex = (currentIndex + 1) % colors.length;
+    });
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _fillStreamColors(singleLink: any, gradientUrl: string): string {
+    if (!this.state.selectedState) {
+      return INACTIVE_FILL_COLOR;
+    } else if (
+      this.state.selectedState &&
+      this.state.selectedLinks.indexOf(singleLink) !== -1 &&
+      this.state.selectedNode
+    ) {
+      return this.state.selectedNode.color;
+    } else if (this.state.selectedState && this.state.selectedLinks.indexOf(singleLink) !== -1) {
+      return gradientUrl;
+    }
+    return INACTIVE_FILL_COLOR;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _fillStreamBorder(singleLink: any, gradientUrl: string): string {
+    if (!this.state.selectedState) {
+      return NON_SELECTED_NODE_AND_STREAM_COLOR;
+    } else if (
+      this.state.selectedState &&
+      this.state.selectedLinks.indexOf(singleLink) !== -1 &&
+      this.state.selectedNode
+    ) {
+      return this.state.selectedNode.borderColor;
+    } else if (this.state.selectedState && this.state.selectedLinks.indexOf(singleLink) !== -1) {
+      return gradientUrl;
+    }
+    return NON_SELECTED_NODE_AND_STREAM_COLOR;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _fillNodeBorder(singleNode: any): string {
+    if (!this.state.selectedState) {
+      return singleNode.borderColor;
+    } else if (
+      this.state.selectedState &&
+      this.state.selectedNodes.indexOf(singleNode) !== -1 &&
+      this.state.selectedNode
+    ) {
+      return this.state.selectedNode.borderColor;
+    } else if (this.state.selectedState && this.state.selectedNodes.indexOf(singleNode) !== -1) {
+      return singleNode.borderColor;
+    }
+    return singleNode.borderColor;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private _getSelectedNodes(selectedLinks: any[]): any[] {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const nodes: any = [];
+    selectedLinks.forEach(link => {
+      nodes.push(link.target);
```

Comment:
Can this have duplicates

Reply:
I have added a check to prevent the duplicates.